code/game/objects/structures/flora.dm

```diff
@@ -133,24 +130,30 @@
 	if(use_default_sound)
 		return ..()
 
-/*
+/**
+ * A helper proc for getting the products that could be created when harvesting this flora, list syntax is (type = weight)
+ * Because of how this works, it can spawn in anomalies if you want it to. Or wall girders
+ * Returns: An assoc list of obj typepaths and their weights e.g. list(/obj/item/food/grass = 1), or null
+ */
+/obj/structure/flora/proc/get_potential_products()
+	return null
+
+/**
  * A helper proc for getting a random amount of products, associated with the flora's product list.
- * Returns: A list where each value is (product_type = amount_of_products)
+ * Returns: A list where each value is (harvested_item_typepath = amount_of_products)
  */
 /obj/structure/flora/proc/get_products_list()
-	if(!LAZYLEN(product_types))
-		return list()
-	var/list/product_list = list()
+	var/list/product_list = get_potential_products()
 
 	var/harvest_amount = rand(harvest_amount_low, harvest_amount_high)
 	for(var/iteration in 1 to harvest_amount)
-		var/chosen_product = pick_weight(product_types)
+		var/chosen_product = pick_weight(product_list)
```

Comment:
Probably shouldn't re-use the list containing weights for the list containing results

Reply:
Whoops, I no longer possess the ability to read anything containing the word 'product' properly